data_tool/create.sh

```diff
@@ -0,0 +1,152 @@
+# Show the usage screen, since no arguments were given
+if [[ $# -eq 0 ]] || [[ $1 == "--help" ]] || [[ $1 == "-h" ]]; then
+    echo "usage: surround data create [-h] [-f FILE|-d DIRECTORY] [-o OUTPUT_FILE]"
+    echo ""
+    echo "Create a data container from a file or directory"
+    echo ""
+    echo "optional arguments:"
+    echo "  -h, --help          show this help message and exit"
+    echo "  -f FILE, --file FILE"
+    echo "                      Path to file to import into container"
+    echo "  -d DIRECTORY, --directory DIRECTORY"
+    echo "                      Path to directory to import into container"
+    echo "  -o OUTPUT, --output OUTPUT"
+    echo "                      Path to file to export container to"
+
+    exit 0
+fi
+
+# Get input file from user input (required)
+if [[ $1 == "--file" ]] || [[ $1 == "-f" ]]; then
+    if [[ $# -lt 2 ]]; then
+        echo "--file argument requires a value"
+        exit -1
+    fi
+
+    INPUT_FILE=$2
+elif [[ $1 == "--directory" ]] || [[ $1 == "-d" ]]; then
+    if [[ $# -lt 2 ]]; then
+        echo "--directory argument requires a value"
+        exit -1
+    fi
+
+    INPUT_DIR=$2
+else
+    echo "--file or --directory argument are required!"
+fi
+
+OUTPUT="./output.data.zip"
+
+# Get output file from user input (no required)
+if [[ $3 == "--output" ]] || [[ $3 == "-o" ]]; then
+    if [[ $# -lt 4 ]]; then
+        echo "--output requires a value"
+    fi
+
+    OUTPUT=$4
+fi
+
+function prompt() {
+    echo "$1"
+    read
+
+    if [[ $# -ge 3 ]] && [[ -z $REPLY ]] && [[ -n $2 ]]; then
+        echo "$2"
+        echo "Enter ? for help on how to answer the field."
+        echo ""
+        prompt "$1" "$2" "$3"
+        return
+    fi
+
+    if [[ $# -ge 3 ]] && [[ $REPLY == "?" ]]; then
+        echo "$3"
+        echo ""
+        prompt "$1" "$2" "$3"
+        return
+    fi
+
+    echo ""
+}
+
+# ============================== Generate Summary Metadata ============================
+echo "=========[Creating a data container]========="
+
+if [[ -n $INPUT_FILE ]]; then
+    echo "Generating metadata for file: $INPUT_FILE"
+else
+    echo "Generating metadata for directory: $INPUT_DIR"
+fi
+
+echo ""
+echo "======[Creating summary metadata]==========="
+echo "Enter ? for help on how to answer the field."
+echo ""
+prompt "What is your name: " "This field is required!" "This field is required to identify who created the container later."
+prompt "Give this data a short title describing its contents: " "This field is required!" "Just a short title that will help you identify the data quickly."
+prompt "Give this data a brief description: " "This field is required!" "Provide a brief description of the contents of this data, to help people understand what it is used for etc.."
```

Comment:
`Give this data a short title describing its contents` might be better expressed as `Give this data a meaningful title`. I skimmed the question and ended up writing a description, then having a d'oh moment when the next question asked for a description.